Doc/library/os.rst

```diff
@@ -3897,16 +3897,22 @@ written in Python, such as a mail server's external command delivery program.
 
 .. function:: pidfd_open(pid, flags=0)
 
-   Return a file descriptor referring to the process *pid*.  This descriptor can
-   be used to perform process management without races and signals.  The *flags*
-   argument is provided for future extensions; no flag values are currently
-   defined.
+   Return a file descriptor referring to the process *pid* with *flags* set.
+   This descriptor can be used to perform process management without races
+   and signals.
 
    See the :manpage:`pidfd_open(2)` man page for more details.
 
    .. availability:: Linux 5.3+
    .. versionadded:: 3.9
 
+   .. data:: PIDFD_NONBLOCK
+
+      This flag indicates that the file descriptor will be non-blocking.
```

Comment:
IMO it's worth it to mention the purpose of the flag, what you wrote in the issue:

"If the process referred to by the file descriptor has not yet terminated, then an attempt to wait on the file descriptor using
[waitid(2)](https://man7.org/linux/man-pages/man2/waitid.2.html) will immediately return the error EAGAIN rather than blocking."